lib/ExpensiMark.js

```diff
@@ -1,7 +1,11 @@
 import Str from './str';
 import TLD_REGEX from './tlds';
 
-const URL_REGEX = `((https?:\\/\\/(?:www\\.)?)?([-a-z0-9]+?\\.)+(?:${TLD_REGEX})(?:(?:\\?|#|\\/)[-\\w()@:;%\\+.,~#&\\/={}\\[\\]?]*[-\\w()@:;%\\+~#?&\\/={}\\[\\]]|\\/|\\b|(?=_)))`;
+const URL_WEBSITE_REGEX = `(https?:\\/\\/)?((?:www\\.)?[-a-z0-9]+?\\.)+(?:${TLD_REGEX})(?:\\:\\d{2,4}|\\b|(?=_))`;
+const URL_PATH_REGEX = '(?:\\/[-\\w$@.&+!*"\'(),=%]*[-\\w~@:%)]|\\/)*';
+const URL_PARAM_REGEX = '\\?[-\\w$@.&+!*"\'(),=%{}:;\\[\\]]+';
+const URL_FRAGMENT_REGEX = '#[-\\w$@.&+!*"\'(),=%;\\/]*';
+const URL_REGEX = `(${URL_WEBSITE_REGEX}${URL_PATH_REGEX}(?:${URL_PARAM_REGEX}|${URL_FRAGMENT_REGEX})?)`;
```

Comment:
You can have a url with both a query param and a fragment, so I think this last bit should be 
`(?:${URL_PARAM_REGEX}?${URL_FRAGMENT_REGEX}?)